components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java

```
@@ -0,0 +1,251 @@
+import org.wso2.carbon.identity.oauth.uma.resource.service.exceptions.UMAException;
+import org.wso2.carbon.identity.oauth.uma.resource.service.exceptions.UMAServiceException;
+import org.wso2.carbon.identity.oauth.uma.resource.service.model.Resource;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.ws.rs.core.Response;
+
+/**
+ * ResourceRegistrationApiServiceImpl is used to handling resource management.
+ */
+public class ResourceRegistrationApiServiceImpl extends ResourceRegistrationApiService {
+
+    private static final Log log = LogFactory.getLog(ResourceRegistrationApiServiceImpl.class);
+
+    /**
+     * Register a resource with resource details
+     *
+     * @param requestedResource details of the resource to be registered
+     * @return Response with the status of the registration
+     */
+    @Override
+    public Response registerResource(ResourceDetailsDTO requestedResource) {
+
+        if (requestedResource == null) {
+            log.error("Request body cannot be empty.");
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+        try {
+            Resource registerResource = ResourceUtils.getResourceService()
+                    .registerResource(ResourceUtils.getResource(requestedResource));
+            CreateResourceDTO createResourceDTO = ResourceUtils.createResponse(registerResource);
+            return Response.status(Response.Status.CREATED).entity(createResourceDTO).build();
+        } catch (UMAServiceException e) {
+            handleErrorResponse(e, log);
+        } catch (UMAClientException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Client error when registering resource in resource server.", e);
+            }
+            handleErrorResponse(e, log);
+        } catch (Throwable throwable) {
+            log.error("Internal server error occurred. ", throwable);
```

Don't we need to return an error message?

---

Returned error message.